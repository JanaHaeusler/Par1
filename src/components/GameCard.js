import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

GameCard.propTypes = {
    location: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    players: PropTypes.string.isRequired,
    winner: PropTypes.string.isRequired,
    shots: PropTypes.string.isRequired,
  }

export default function GameCard({location, date, players, winner, shots}) {
 return(
    <CardWrapper>
        <Location>{location}</Location>
        <Date>{date}</Date>
       
            <HeadlinePlayers>Player(s)</HeadlinePlayers>
            <DataPlayers>{players}</DataPlayers>
      
            <HeadlineWinner>Winner(s)</HeadlineWinner>
            <DataWinner>{winner}</DataWinner>
        
       
            <HeadlineShots>Total Shots</HeadlineShots>
            <DataShots>{shots}</DataShots>
    </CardWrapper>
 )
}

const CardWrapper = styled.div`
    padding: 10px;
    display: grid;
    grid-template-rows: 1fr 1fr 1fr;
    grid-template-columns: 1fr 1fr 1fr;
    row-gap: 10px;
    column-gap: 10px;
    justify-items: center;
    box-shadow: 0 0 10px var(--primary-medium);
    border-radius: 10px;
    background-color: var(--primary);
    font-size: 0.9rem;
    text-align: center;
    

    h4 {
        margin: 0;
        font-weight: 550;
    }
`
const Location = styled.div`
    grid-column: 1 / 3;
    justify-self: start;
`
const Date = styled.div`
    justify-self: end;
`
const HeadlinePlayers = styled.h4`
    place-items: center;
`
const DataPlayers = styled.span`
    grid-column: 1 / 2;
    grid-row: 3 / 4;
    place-items: center;
`
const HeadlineWinner = styled.h4`
    place-items: center;
`
const DataWinner = styled.span`
    grid-column: 2 / 3;
    grid-row: 3 / 4;
    place-items: center;
`
const HeadlineShots = styled.h4`
    place-items: center;
`
const DataShots = styled.span`
    place-items: center;
`
