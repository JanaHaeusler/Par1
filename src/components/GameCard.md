```js
<GameCard
  location="Horner Racecourse"
  date="11/04/2020"
  players="John, Jane"
  winner="Jane"
  shots="56"
/>
```
