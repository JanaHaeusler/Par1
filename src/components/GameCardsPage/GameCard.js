import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import {useState} from 'react'
import {Bin, Pencil, Cross} from '../Icons'
import ButtonPrimary from '../buttons/ButtonPrimary'
import ButtonSecondary from '../buttons/ButtonSecondary'

GameCard.propTypes = {
    location: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    players: PropTypes.string.isRequired,
    winner: PropTypes.string.isRequired,
    shots: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    onDelete: PropTypes.func.isRequired,
    onEdit: PropTypes.func.isRequired,
    showSaveGamePage: PropTypes.func.isRequired,
}

export default function GameCard({location, date, players, winner, shots, id, onDelete, onEdit, showSaveGamePage}) {
 
    const [isSetToDelete, setIsSetToDelete] = useState(false)

    return(
        <Card>
            {isSetToDelete || (
               <> 
                    <SavedGameContent>
                        <Location>{location}</Location>
                        <Date>{date}</Date>
                        <PlayerWrapper>
                            <h4>Player(s)</h4>
                            <span>{players}</span>
                        </PlayerWrapper>
                        <WinnerWrapper>
                            <h4>Winner(s)</h4>
                            <span>{winner}</span>
                        </WinnerWrapper>
                        <ShotsWrapper>
                            <h4>Total Shots</h4>
                            <span>{shots}</span>
                        </ShotsWrapper>
                    </SavedGameContent>
                    <ButtonWrapper>
                            <ButtonDeleteIcon onClick={() => setIsSetToDelete(true)} data-testid="button-delete-icon"><BinIcon/></ButtonDeleteIcon>
                            <ButtonEditIcon onClick={() => handleEdit(id)} data-testid="button-edit-icon"><PenIcon/></ButtonEditIcon>
                    </ButtonWrapper>
                </>
            )}
            {isSetToDelete && (
                <DeleteField>
                    <span>Do you want to delete this gamecard?</span>
                    <ButtonDelete onClick={() => onDelete(id)} data-testid="button-delete"><BinIcon/>Delete</ButtonDelete>
                    <ButtonCancel onClick={() => setIsSetToDelete(false)} data-testid="button-cancel"><CancelIcon/>Cancel</ButtonCancel>
                </DeleteField>
            )}
        </Card>
    )

    function handleEdit(id) {
        showSaveGamePage()
        onEdit(id)
    }
}

const Card = styled.section`
    margin: 0 20px;
    padding: 10px;
    box-shadow: 0 0 10px var(--primary-medium);
    border-radius: 10px;
    background-color: var(--text-light);
    font-size: 0.9rem;
`
const SavedGameContent = styled.div`
    display: grid;
    grid-template-rows: repeat(4, auto);
    grid-template-columns: 1fr 1fr;
    grid-gap: 10px;
`
const Date = styled.div`
    grid-column-start: 2;
    grid-row-start: 1;
    justify-self: end;
`
const Location = styled.div`
    grid-column: 1 / 3;
    grid-row-start: 2; 
    text-transform: uppercase;
`
const PlayerWrapper = styled.div`
    grid-column: 1 / 4;
    grid-row-start: 3;
`
const WinnerWrapper = styled.div`
    grid-column-start: 1;
    grid-row-start: 4;
`
const ShotsWrapper = styled.div`
    grid-column-start: 2;
    grid-row-start: 4;
`
const ButtonWrapper = styled.div`
    display: flex;
    justify-content: space-evenly;
    align-items: baseline;
    width: 100%;
`
const ButtonDeleteIcon = styled.button`
    display: flex;
    margin: 5px;
    padding: 0;
    width: 30px;
    border: none;
    background: none;
`
const BinIcon = styled(Bin)`
    margin-right: 3px;
    fill: var(--secondary-dark);
`
const ButtonEditIcon = styled.button`
    display: flex;
    margin: 20px 5px 5px 5px;
    padding: 0;
    width: 30px;
    border: none;
    background: none;
`
const PenIcon = styled(Pencil)`
    margin-right: 3px;
    fill: var(--secondary-dark);
`
const DeleteField = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 1fr 1fr;
    gap: 10px 20px;
    place-items: center;
        
    span{
        grid-column: 1 / 3;
        text-align: center;
        color: var(--secondary-dark);
        font-weight: 800; 
    }
`
const ButtonDelete = styled(ButtonSecondary)`
    grid-column-start: 1;
    grid-row-start: 2;
`
const ButtonCancel = styled(ButtonPrimary)`
    grid-column-start: 2;
    grid-row-start: 2;
`
const CancelIcon = styled(Cross)`
    margin-right: 3px;
    fill: var(--text-light);
`