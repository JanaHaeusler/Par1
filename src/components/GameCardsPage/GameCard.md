```jsx
<GameCard
  location="Horner Racecourse"
  date="11/04/2020"
  players="John, Jane"
  winner="Jane"
  shots="56"
  id="1"
  onDelete={() => {}}
  onEdit={() => {}}
  showSaveGamePage={() => {}}
/>
```
