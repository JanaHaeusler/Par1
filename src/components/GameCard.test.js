import GameCard from './GameCard'
import { render } from '@testing-library/react'

describe('GameCard', () => {
  it('renders correctly', () => {
    const { container } = render(
        <GameCard 
            location="Horner Racecourse" 
            date="11/04/2020"
            players="John, Jane"
            winner="Jane"
            shots="56"
        />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows all texts', () => {
    const { getByText } = render(
        <GameCard 
            location="Horner Racecourse" 
            date="11/04/2020"
            players="John, Jane"
            winner="Jane"
            shots="56"
        />)
    expect(getByText('Horner Racecourse')).toBeInTheDocument()
    expect(getByText('11/04/2020')).toBeInTheDocument()
    expect(getByText('John, Jane')).toBeInTheDocument()
    expect(getByText('Jane')).toBeInTheDocument()
    expect(getByText('56')).toBeInTheDocument()
  })
})