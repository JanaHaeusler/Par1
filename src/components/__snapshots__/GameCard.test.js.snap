// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard renders correctly 1`] = `
.c0 {
  padding: 10px;
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
  grid-template-columns: 1fr 1fr 1fr;
  row-gap: 10px;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  justify-items: center;
  box-shadow: 0 0 10px var(--primary-medium);
  border-radius: 10px;
  background-color: var(--primary);
  font-size: 0.9rem;
  text-align: center;
}

.c0 h4 {
  margin: 0;
  font-weight: 550;
}

.c1 {
  grid-column: 1 / 3;
  justify-self: start;
}

.c2 {
  justify-self: end;
}

.c3 {
  place-items: center;
}

.c4 {
  grid-column: 1 / 2;
  grid-row: 3 / 4;
  place-items: center;
}

.c5 {
  place-items: center;
}

.c6 {
  grid-column: 2 / 3;
  grid-row: 3 / 4;
  place-items: center;
}

.c7 {
  place-items: center;
}

.c8 {
  place-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Horner Racecourse
  </div>
  <div
    class="c2"
  >
    11/04/2020
  </div>
  <h4
    class="c3"
  >
    Player(s)
  </h4>
  <span
    class="c4"
  >
    John, Jane
  </span>
  <h4
    class="c5"
  >
    Winner(s)
  </h4>
  <span
    class="c6"
  >
    Jane
  </span>
  <h4
    class="c7"
  >
    Total Shots
  </h4>
  <span
    class="c8"
  >
    56
  </span>
</div>
`;
